src/zope/traversing/api.py

```diff
@@ -185,30 +183,29 @@ def canonicalPath(path_or_object):
 
     See `ITraversalAPI` for details.
     """
-    if isinstance(path_or_object, six.string_types):
+    if isinstance(path_or_object, str):
         path = path_or_object
         if not path:
             raise ValueError("path must be non-empty: %s" % path)
     else:
         path = getPath(path_or_object)
 
-    path = u'' + path
+    path = '' + path
 
     # Special case for the root path.
-    if path == u'/':
+    if path == '/':
         return path
 
-    if path[0] != u'/':
+    if path[0] != '/':
         raise ValueError('canonical path must start with a "/": %s' % path)
-    if path[-1] == u'/':
+    if path[-1] == '/':
         raise ValueError('path must not end with a "/": %s' % path)
 
     # Break path into segments. Process '.' and '..' segments.
     return _normalizePath(path)
 
 
 # import this down here to avoid circular imports
-from zope.traversing.adapters import traversePathElement
```

Comment:
So the circular import problem is gone?  Anyway, please remove the comment too.

Reply:
Fixed in https://github.com/zopefoundation/zope.traversing/pull/19/commits/92c8287b5a247fa30741ba71d05a2ded4c1261d6.